src/services/__tests__/newMessagesService.test.ts

```diff
@@ -823,20 +841,154 @@ describe("MessageService#getThirdPartyMessage", () => {
     );
 
     const res = await service.getThirdPartyMessage(
-      aValidMessageWithThirdPartyData
+      {
+        ...aValidMessage,
+        content: {
+          ...aValidMessage.content,
+          third_party_data: { 
+            ...aValidThirdPartyData, 
+            has_attachments: true
+          },
+        }
+      } as unknown as MessageWithThirdPartyData
+    );
+    
+    expect(mockGetTPMessageFromExternalService).toHaveBeenCalledWith({
+      id: aValidThirdPartyMessageUniqueId,
+    });
+    expect(res).toMatchObject({
+      kind: "IResponseErrorBadGateway",
+      detail: "Bad Gateway: ATTACHMENTS_NOT_PRESENT"
+    });
+  });
+
+  it("should return error Bad Gateway REMOTE_CONTENT_NOT_PRESENT if ThirParty service client does not return the remote content when the flag has_remote_content is true", async () => {
+    mockGetTPMessageFromExternalService.mockImplementationOnce(async (_messageId) =>
+      t.success({
+        status: 200,
+        headers: {},
+        value: {
+          ...aThirdPartyMessage,
+          details: undefined
+        },
+      })
+    );
+    const service = new NewMessageService(
+      api,
+      mockGetThirdPartyMessageClientFactory
+    );
+
+    const res = await service.getThirdPartyMessage(
+      {
+        ...aValidMessage,
+        content: {
+          ...aValidMessage.content,
+          third_party_data: { 
+            ...aValidThirdPartyData, 
+            has_remote_content: true
+          },
+        }
+      } as unknown as MessageWithThirdPartyData
     );
 
     expect(mockGetTPMessageFromExternalService).toHaveBeenCalledWith({
       id: aValidThirdPartyMessageUniqueId,
     });
     expect(res).toMatchObject({
-      kind: "IResponseErrorInternal",
+      kind: "IResponseErrorBadGateway",
+      detail: "Bad Gateway: REMOTE_CONTENT_NOT_PRESENT"
     });
   });
 
-  it("should return Not Found if ThirParty service client returns an 404", async () => {
-    mockGetTPMessageFromExternalService.mockImplementationOnce(async () =>
-      t.success({ status: 404 })
+  it("should return error Bad Gateway MARKDOWN_VALIDATION_ERROR if ThirParty service client returns an invalid markdown when the flag has_remote_content is true", async () => {
+    mockGetTPMessageFromExternalService.mockImplementationOnce(async (_messageId) =>
+      t.success({
+        status: 200,
+        headers: {},
+        value: {
+          ...aThirdPartyMessage,
+          details: {
+            ...aThirdPartyMessage.details,
+            markdown: "invalid"
```

Comment:
Maybe we should explain WHY it's invalid. I know that it should be over 80 chars, BUT other devs could have to search this "insight".

Reply:
Maybe we can refactor by extracting a `const invalidMarkdown = 'invalid'` and comment above that a valid markdown must be between 80 and 10000 chars. Or so.